andino_firmware/src/app.cpp

```diff
@@ -123,47 +122,32 @@ void App::setup() {
 
   left_pid_controller_.reset(left_encoder_.read());
   right_pid_controller_.reset(right_encoder_.read());
+
+  // Initialize command shell.
+  shell_.init(Serial);
+  shell_.set_default_callback(cmd_unknown_cb);
+  shell_.register_command(Commands::kReadAnalogGpio, cmd_read_analog_gpio_cb);
+  shell_.register_command(Commands::kReadDigitalGpio, cmd_read_digital_gpio_cb);
+  shell_.register_command(Commands::kReadEncoders, cmd_read_encoders_cb);
+  shell_.register_command(Commands::kReadHasImu, cmd_read_has_imu_cb);
+  shell_.register_command(Commands::kReadEncodersAndImu, cmd_read_encoders_and_imu_cb);
+  shell_.register_command(Commands::kResetEncoders, cmd_reset_encoders_cb);
+  shell_.register_command(Commands::kSetMotorsSpeed, cmd_set_motors_speed_cb);
+  shell_.register_command(Commands::kSetMotorsPwm, cmd_set_motors_pwm_cb);
+  shell_.register_command(Commands::kSetPidsTuningGains, cmd_set_pid_tuning_gains_cb);
+  /* Initialise the IMU sensor */
+  if(!bno_.begin())
+  {
+    /* There was a problem detecting the BNO055 ... check your connections */
+    Serial.print("Ooops, no BNO055 detected ... Check your wiring or I2C ADDR!");
```

Comment:
Unfortunately, there is no support yet for a safe logging via serial port (so as to prevent communication mishaps with the SBC). Let's please keep this as a comment and feel free to open a ticket or send a PR to add logging support.